File: Java/RecyclerView11BaseClassForActivityAdapter/app/src/main/java/dev/ogabek/recyclerview11_baseclassforactivityadapter/adapter/BaseAdapter.java
package dev.ogabek.recyclerview11_baseclassforactivityadapter.adapter;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import dev.ogabek.recyclerview11_baseclassforactivityadapter.network.BaseServer;

public class BaseAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> implements BaseServer {
    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return null;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return 0;
    }
}

File: Kotlin/RecyclerView11BaseClassForActivityAndAdapter/app/src/main/java/dev/ogabek/recyclerview_11_baseclassforactivityandadapter/adapter/BaseAdapter.kt
package dev.ogabek.recyclerview_11_baseclassforactivityandadapter.adapter

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import dev.ogabek.recyclerview_11_baseclassforactivityandadapter.network.BaseServer

open class BaseAdapter: RecyclerView.Adapter<RecyclerView.ViewHolder>(), BaseServer {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        TODO("Not yet implemented")
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        TODO("Not yet implemented")
    }

    override fun getItemCount(): Int {
        TODO("Not yet implemented")
    }

}